src/wallet/wallet2.cpp

```diff
@@ -5025,6 +5025,230 @@ bool wallet2::get_tx_key(const crypto::hash &txid, crypto::secret_key &tx_key) c
   tx_key = i->second;
   return true;
 }
+//----------------------------------------------------------------------------------------------------
+std::vector<std::vector<crypto::signature>> wallet2::get_spend_proof(const cryptonote::transaction_prefix& tx)
+{
+  THROW_WALLET_EXCEPTION_IF(m_watch_only, error::wallet_internal_error,
+    "get_spend_proof requires spend secret key and is not available for a watch-only wallet");
+
+  std::vector<std::vector<crypto::signature>> signatures;
+
+  const crypto::hash tx_prefix_hash = cryptonote::get_transaction_prefix_hash(tx);
+
+  for(size_t i = 0; i < tx.vin.size(); ++i)
+  {
+    if(tx.vin[i].type() != typeid(cryptonote::txin_to_key))
+      continue;
+    const txin_to_key& in_key = boost::get<txin_to_key>(tx.vin[i]);
```

Comment:
FYI if you did `boost::get<txin_to_key>(std::addressof(tx.vin[i]));` it will return a pointer to the element if the type is correct, and `nullptr` otherwise. Little difference in terms of performance, I think.

Reply:
Didn't know this usage, indeed I like it better. Thanks!